src/codegen.zig

```diff
@@ -1270,6 +1272,24 @@ fn Function(comptime arch: std.Target.Cpu.Arch) type {
             return self.finishAir(inst, result, .{ bin_op.lhs, bin_op.rhs, .none });
         }
 
+        fn airShl(self: *Self, inst: Air.Inst.Index) !void {
+            const bin_op = self.air.instructions.items(.data)[inst].bin_op;
+            const result: MCValue = if (self.liveness.isUnused(inst)) .dead else switch (arch) {
+                .arm, .armeb => try self.genArmBinOp(inst, bin_op.lhs, bin_op.rhs, .shl),
```

Comment:
The code for `shl` is not present yet in `genArmBinOpCode`, which is called by `genArmBinOp`. This means that the code would hit the `unreachble` in that switch. It shouldn't be too hard to implement this, but I can also do that in a follow-up PR.

Reply:
This is more complex than anticipated, new psedo-instructions will have to be added to `codegen/arm.zig`. I'll do this in a follow-up PR.